docs/notebooks/longrangecnot/LongRangeCNOTgates.ipynb

```diff
@@ -0,0 +1,2139 @@
aJu/S7UEL1zmFGN+jdBprg2Y5+VyNBvFIr8LQPh8pXnY4XaL4rvql+L+DN9btAERgFes2/pb/K5NO8CKQ5adf2qb9EYz6M4g1XMRf3jKvDG/A3lkk8ynennoFXXz6hJfhBh+hmK/C0jw/V5/Ov+84fR38FvPk04bfBVa8m82loS6F787Qo4hECBGdzUqP2gT9H99TjRC1Xn74Er4PRJ/1i+iOfkqOTAgjFNLIDRvUzSG5W8mHMt8/HjWM/0aIe81Fyz6jf3QOXDTJJuPLbpvwFeH364A4Z/Ia1BEa9S8GFPmiAUbcET6FoB2Mx3rjh41t53PNifvLhdE/4sd88TRkORoLwC3D4/LKD5vNZdpHlRTzdgF9wKvHifrs+aF2ebug/wpdKw5tthjfBzt32IbvshDZ26ac6YmDV2XWTaBPxWU9fHbe2DIYP+0HSrife98LT40xlstKUAfUU4s+/fDerg9MafG61+skXMef4qjia/5qSEJUzSmVNlJ2UONO16o/Jv9vp2bcsg+kYDjOqUdtwNLhOyl/yMkyEJmT693a5KoN/s4Fr/Rv7dnXvU07jVICpmODUGUr+K07RAZ9uqA07iF65/CQYfWorNNJ/tC/Xhvap8kYM7sMoHAndZNHHf8VdRZcFHDr4fVB2qlsw6oYa0RIY9bIK1vt+7XxVKsDkDCt+NjCjnokxaZJjVOUrdGGjEpST5MNIRhXa3NTQPr3bqAYJEW45GGlGAxzcc/gEfz78fjMblUCZhfa7B/sLYxOlFwtaUFHMLw5eXhGMjpd2i0fByMdpoIJUbDEeqVDF0VRXnLqHvtTH0Vmhi4rx1kCbTTxz5ThY/H7WjxbytF+OrhSntBOMAl33hYLivtU9tYeLIzG1R99ppzMOuLq2NdCiENASflGhSkKE7ykp1AUMDXTI4wb6g79d0ivkcbLHAcd32uCS4OdeXtwok+7qZktg1MtqX75QBwf4cMnHB48cvYFHJ2MEB6+Jxx0MgkXF6KoXhT8fLP125euqRunTzOQKFXxCFF4Jla17Apvk5Ioe83EwqT/1nI9rTyGf8roiKMbp8HwW/iqDCuXiiEXXZib3roBQpVxUh+6pWK5ObuuxPiTHQ+6dAsz52+5d96zw29Ghg19/+LhC3yGeFuiCr9BFxTzt2lOAyVdz/vMdMKqxhUwt5Wn3fgFJPq50T0VDd6rbwbgHnnZyEhpwdEG+Yhm6o5+Fz5bwi98+AQdghUuYmL0UttDAyGybnH7Clm78qtWmWLFJmFScqlIYeRG8wrV4rvsxiSTjasbbWTuJcJVvjIvY9z5DuMeHcvbJ28wO6pAlZBibivLwvTJ3zfE4uWQNBVQ62mLNn70KFsqMqcHQPDbzJRaiJolLHjcd8RkN1rkQJ5rBMJNrVIQVx/3c8Nz1ljCLay7M9JaSw27+6z7/VcHS8Joaa5giFILzSTSbYfWg3PZxYko0s+IawiWzDtiZOpD73k0hMAXh3NM7Xs9nZlPVhk3WWgwjRWl1g4JbnJG1UyWWBhYP16xglF3bAON6YMRlLxjlNpXbN8KUwg48ktchmnlxt1QanO78UBn2nkThaptXzUfEKbwWYCwwgetHEBoHT/5j4VfwpZnC0RSIEzY7N21H3WHi19QBDoe7y7Qj+87fqMjBCFflgNGXZn4mwQjYSdzLwo/cxkoOPvo4DR6FQ02HSGhoCsFtg7QTGGoX0x0NEnOnh668Pf/xYQQQt/fOzrkFS47prBR9ram/KNOUDTXwTKQh93iUKa4oSxx37kYyuDYzrerwCKxO+u1cx15/CQhZJFpaz0kM6mpNDWrqSsJLTReehD95DASrYNZ0jarXFKCjT74Lp3rPn6KJQ5vEDFBuK/K0ERvB9gcORqZZnJIFRk0nxNXH/Jcr30eWm1Yq0KemHrSNjKaehCqRj6a64nSx56YT4Gv1QCvwtMSPVlU31CUYmfIXjKrb8Y6m5oBFclJTcdL3CllQm5TX6W/yRlklp+lDFsrzABksxO9MDNzbh9SIp7O18OpW5IwUE0Wqr1WPaFHkoOlW4VX97UIngEM0kSIGxdUPDeiZaKBBBolfdBzN/vK05A7LQDPEBeUY+WsKXfiQItWUr+DMgt80fepgRFY2qXPAX5Kp0Ch4dzhUEHmr8DTTTh7yG9yJFrPU7/iF4jXlqhhL4UTT2ZqqdDDCG5LnDj28p6l4x/C8I76TDAonKIcp9dbhaQLBHE8X+GV3PE3YjqYBRR6OpzUNq++OTkUfPAP3IrmE9BFTX63D05IhzF+yesurA1dMV2rqdPc8TVaAEH6dzhGL8MmweCTLCl0niwS3RtZqSKvpaYVd1DtQM1sJcunSiSro41zW2pdssnlL0qy0NRtcFbbUg0zps+qyb4+wPTMpbUOhAa22ncwKUXuJxSK08cpPkJfntl0hBDXIA+gI2woDi5zMW3vbF3MGVX8qxUqDQnrSE4/FT7A37XBrPyZi55wXt/UYMH++m+0SIBwJPdk9Wga6kanrS3+asG0w5r0EzimOT92VJzTidOpXsBLwBl6gru6f4iV8ak4aKtNukgpwioLrusfor1eIGLyaudKR4BkqUVJncEkjIAaeELYzCdqcOz9rjz+UdjC27x2yz/2YIFgUyNuLcnb8UTFbgrC7744UhmOhz1QPHO4R3Ogt+Rn7EBFA1+XDukkGdShfd5ccjPS+JNea+7jWWqjP5QSj9+a7RvX5l9UMwTj+vIideHzcXnw1ZS/9CyOB9ztVhewTn0jYZhh64RKWnrOaaibzuo/+KW3l6DAg2IHHja+at4L9OyrPJxD8WB6Chz3B1/AuEiy1gdCE2+iCvpze8DWkbsd8HyiPygAWbRVw/JejNp4g4388zqIBgnJL6MsUPN1jfNg+9jECUJ9P22RWb5Wit5w+EA6UXF/BTOv/RajFrzk88+usWDkIGCGWvcJIITpip24JMP6SULPTLNSDNkrxFBSjitfXMEbJGV+KWe9eYfsziwLWLcy5FW
```

Comment:
There's enough methods to create a library. Shouldn't we add these utils separately and avoid having them defined inside the notebook?

Reply:
Yes. This was the original plan. This should be done later once a solid module is developed. I separate rate notebook can go with this module on state certification and measuring average gate fidelities.

Reply:
Sounds good! I'll put a plan in place for the future development of its own module. Let's keep it in the notebook for our first iteration